src/prpy/util.py

```diff
@@ -1326,7 +1326,7 @@ def ComputeGeodesicUnitTiming(traj, env=None, alpha=1.0):
     return new_traj
 
 
-def CheckJointLimits(robot, q, deterministic=None):
+def CheckJointLimits(robot, q, hard_limits, deterministic=None):
```

Comment:
This type of logic does seem robot-specific though -- perhaps it would make more sense to put this sort of logic in `herbpy` instead, in the `Planner` wrapper? The wrapper could simply extend the joint limits temporarily (up to the hard limits) before invoking the planner. That way each planner would still only have to know about one set of limits.